pdftopng causes installation issues for M1 Mac and Python 3.10

**Question**

Hi there,

First of all, a huge thank you for developing this and for maintaining it the way you have (and the tutorials are fantastic!). I want to use `ehrapy` for my epidemiological clinical data and have been trying to install directly via `pip install`. Unfortunately, this does not work for me as it gives the output found below. 

**Environment**

pip version: 22.3.1
Python version: 3.9.15
OS: MacOS 

**Description**

`pip` is unable to install the package `ehrapy`
It seems like it cannot install because I don't have `pdftopng` installed. Unfortunately, I am having issues installing it on my machine and it has already been raised [here](https://github.com/vinayak-mehta/pdftopng/issues/11). 

-- Is it possible to install `ehrapy` without `pdftopng`?

```
Collecting ehrapy
  Using cached ehrapy-0.2.0-py3-none-any.whl (131 kB)
Collecting questionary>=1.10.0
  Using cached questionary-1.10.0-py3-none-any.whl (31 kB)
Collecting fancyimpute>=0.7.0
  Using cached fancyimpute-0.7.0.tar.gz (25 kB)
  Preparing metadata (setup.py) ... done
Requirement already satisfied: requests>=2.26.0 in /opt/miniconda3/envs/tuberculosis/lib/python3.9/site-packages (from ehrapy) (2.28.1)
Requirement already satisfied: anndata>=0.7.8 in /opt/miniconda3/envs/tuberculosis/lib/python3.9/site-packages (from ehrapy) (0.8.0)
Collecting pypi-latest>=0.1.1
  Using cached pypi_latest-0.1.2-py3-none-any.whl (10 kB)
Collecting leidenalg>=0.8.7
  Using cached leidenalg-0.9.0-cp39-cp39-macosx_11_0_arm64.whl (202 kB)
Collecting lifelines<0.28.0,>=0.27.0
  Using cached lifelines-0.27.4-py3-none-any.whl (349 kB)
Requirement already satisfied: PyYAML>=5.4.1 in /opt/miniconda3/envs/tuberculosis/lib/python3.9/site-packages (from ehrapy) (6.0)
Collecting miceforest>=5.3.0
  Using cached miceforest-5.6.2-py3-none-any.whl (57 kB)
Collecting category_encoders>=2.2.2
  Using cached category_encoders-2.5.1.post0-py2.py3-none-any.whl (72 kB)
Requirement already satisfied: pynndescent>=0.5.4 in /opt/miniconda3/envs/tuberculosis/lib/python3.9/site-packages (from ehrapy) (0.5.8)
Collecting rich>=10.12.0
  Using cached rich-12.6.0-py3-none-any.whl (237 kB)
Requirement already satisfied: pandas>=1.3.3 in /opt/miniconda3/envs/tuberculosis/lib/python3.9/site-packages (from ehrapy) (1.4.4)
Collecting mudata
  Using cached mudata-0.2.1-py3-none-any.whl (23 kB)
Requirement already satisfied: scanpy>=1.8.2 in /opt/miniconda3/envs/tuberculosis/lib/python3.9/site-packages (from ehrapy) (1.9.1)
Requirement already satisfied: session-info>=1.0.0 in /opt/miniconda3/envs/tuberculosis/lib/python3.9/site-packages (from ehrapy) (1.0.0)
Collecting deepl>=1.2.0
  Using cached deepl-1.11.0-py3-none-any.whl (36 kB)
Collecting pyhpo[all]>=3.0.0
  Using cached pyhpo-3.1.3-py3-none-any.whl (15.8 MB)
Requirement already satisfied: scikit-learn>=1.0 in /opt/miniconda3/envs/tuberculosis/lib/python3.9/site-packages (from ehrapy) (1.1.3)
Requirement already satisfied: ipython>=7.30.1 in /opt/miniconda3/envs/tuberculosis/lib/python3.9/site-packages (from ehrapy) (8.6.0)
Collecting deep-translator>=1.6.1
  Using cached deep_translator-1.9.1-py3-none-any.whl (30 kB)
Collecting camelot-py[base]>=0.10.1
  Using cached camelot_py-0.10.1-py3-none-any.whl (40 kB)
Requirement already satisfied: Jinja2>=3.0.1 in /opt/miniconda3/envs/tuberculosis/lib/python3.9/site-packages (from ehrapy) (3.1.2)
Collecting scikit-misc>=0.1.4
  Using cached scikit_misc-0.1.4-cp39-cp39-macosx_11_0_arm64.whl (8.1 MB)
Requirement already satisfied: click>=7.0.0 in /opt/miniconda3/envs/tuberculosis/lib/python3.9/site-packages (from ehrapy) (8.1.3)
Requirement already satisfied: packaging>=20 in /opt/miniconda3/envs/tuberculosis/lib/python3.9/site-packages (from anndata>=0.7.8->ehrapy) (21.3)
Requirement already satisfied: natsort in /opt/miniconda3/envs/tuberculosis/lib/python3.9/site-packages (from anndata>=0.7.8->ehrapy) (8.2.0)
Requirement already satisfied: scipy>1.4 in /opt/miniconda3/envs/tuberculosis/lib/python3.9/site-packages (from anndata>=0.7.8->ehrapy) (1.9.3)
Requirement already satisfied: numpy>=1.16.5 in /opt/miniconda3/envs/tuberculosis/lib/python3.9/site-packages (from anndata>=0.7.8->ehrapy) (1.23.3)
Requirement already satisfied: h5py>=3 in /opt/miniconda3/envs/tuberculosis/lib/python3.9/site-packages (from anndata>=0.7.8->ehrapy) (3.7.0)
Requirement already satisfied: PyPDF2>=1.26.0 in /opt/miniconda3/envs/tuberculosis/lib/python3.9/site-packages (from camelot-py[base]>=0.10.1->ehrapy) (2.11.1)
Requirement already satisfied: tabulate>=0.8.9 in /opt/miniconda3/envs/tuberculosis/lib/python3.9/site-packages (from camelot-py[base]>=0.10.1->ehrapy) (0.9.0)
Requirement already satisfied: chardet>=3.0.4 in /opt/miniconda3/envs/tuberculosis/lib/python3.9/site-packages (from camelot-py[base]>=0.10.1->ehrapy) (5.1.0)
Requirement already satisfied: pdfminer.six>=20200726 in /opt/miniconda3/envs/tuberculosis/lib/python3.9/site-packages (from camelot-py[base]>=0.10.1->ehrapy) (20221105)
Requirement already satisfied: openpyxl>=2.5.8 in /opt/miniconda3/envs/tuberculosis/lib/python3.9/site-packages (from camelot-py[base]>=0.10.1->ehrapy) (3.0.10)
INFO: pip is looking at multiple versions of anndata to determine which version is compatible with other requirements. This could take a while.
Collecting anndata>=0.7.8
  Using cached anndata-0.8.0-py3-none-any.whl (96 kB)
INFO: pip is looking at multiple versions of <Python from Requires-Python> to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of ehrapy to determine which version is compatible with other requirements. This could take a while.
Collecting ehrapy
  Using cached ehrapy-0.1.0-py3-none-any.whl (104 kB)
Collecting medcat>=1.2.6
  Using cached medcat-1.5.0-py3-none-any.whl (141 kB)
Collecting anndata<0.8.0,>=0.7.6
  Using cached anndata-0.7.8-py3-none-any.whl (91 kB)
ERROR: Could not find a version that satisfies the requirement pdftopng>=0.2.3; extra == "base" (from camelot-py[base]) (from versions: 0.1.0)
ERROR: No matching distribution found for pdftopng>=0.2.3; extra == "base"
```


For reference, this is the error when I attempt to install `pdftopng` using `pip`:

```
Collecting pdftopng
  Using cached pdftopng-0.1.0.tar.gz (11 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: Click>=7.0 in /opt/miniconda3/envs/tuberculosis/lib/python3.9/site-packages (from pdftopng) (8.1.3)
Building wheels for collected packages: pdftopng
  Building wheel for pdftopng (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for pdftopng (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [35 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.macosx-11.0-arm64-cpython-39
      creating build/lib.macosx-11.0-arm64-cpython-39/pdftopng
      copying src/pdftopng/__init__.py -> build/lib.macosx-11.0-arm64-cpython-39/pdftopng
      copying src/pdftopng/__version__.py -> build/lib.macosx-11.0-arm64-cpython-39/pdftopng
      running egg_info
      writing src/pdftopng.egg-info/PKG-INFO
      writing dependency_links to src/pdftopng.egg-info/dependency_links.txt
      writing requirements to src/pdftopng.egg-info/requires.txt
      writing top-level names to src/pdftopng.egg-info/top_level.txt
      reading manifest file 'src/pdftopng.egg-info/SOURCES.txt'
      writing manifest file 'src/pdftopng.egg-info/SOURCES.txt'
      copying src/pdftopng/pdftopng.cpp -> build/lib.macosx-11.0-arm64-cpython-39/pdftopng
      running build_ext
      creating var
      creating var/folders
      creating var/folders/gr
      creating var/folders/gr/kg19m5ys29q66kj3f5r4hz2xz29w56
      creating var/folders/gr/kg19m5ys29q66kj3f5r4hz2xz29w56/T
      clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /opt/miniconda3/envs/tuberculosis/include -arch arm64 -fPIC -O2 -isystem /opt/miniconda3/envs/tuberculosis/include -arch arm64 -I/opt/miniconda3/envs/tuberculosis/include/python3.9 -c /var/folders/gr/kg19m5ys29q66kj3f5r4hz2xz29w56/T/tmpbv05lvi2.cpp -o var/folders/gr/kg19m5ys29q66kj3f5r4hz2xz29w56/T/tmpbv05lvi2.o -std=c++14
      clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /opt/miniconda3/envs/tuberculosis/include -arch arm64 -fPIC -O2 -isystem /opt/miniconda3/envs/tuberculosis/include -arch arm64 -I/opt/miniconda3/envs/tuberculosis/include/python3.9 -c /var/folders/gr/kg19m5ys29q66kj3f5r4hz2xz29w56/T/tmp1yjxg25b.cpp -o var/folders/gr/kg19m5ys29q66kj3f5r4hz2xz29w56/T/tmp1yjxg25b.o -fvisibility=hidden
      building 'pdftopng.pdftopng' extension
      creating build/temp.macosx-11.0-arm64-cpython-39
      creating build/temp.macosx-11.0-arm64-cpython-39/src
      creating build/temp.macosx-11.0-arm64-cpython-39/src/pdftopng
      clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /opt/miniconda3/envs/tuberculosis/include -arch arm64 -fPIC -O2 -isystem /opt/miniconda3/envs/tuberculosis/include -arch arm64 -DVERSION_INFO=\"0.1.0\" -Ilib/poppler -Ilib/poppler/fofi -Ilib/poppler/goo -Ilib/poppler/poppler -Ilib/poppler/build -Ilib/poppler/build/poppler -Ilib/poppler/utils -Ilib/poppler/build/utils -I/private/var/folders/gr/kg19m5ys29q66kj3f5r4hz2xz29w56/T/pip-build-env-wb_tew_c/overlay/lib/python3.9/site-packages/pybind11/include -I/opt/miniconda3/envs/tuberculosis/include/python3.9 -c src/pdftopng/pdftopng.cpp -o build/temp.macosx-11.0-arm64-cpython-39/src/pdftopng/pdftopng.o -O3 -Wall -shared -fPIC -stdlib=libc++ -mmacosx-version-min=10.7 -std=c++14 -fvisibility=hidden
      clang: warning: argument unused during compilation: '-shared' [-Wunused-command-line-argument]
      src/pdftopng/pdftopng.cpp:44:10: fatal error: 'config.h' file not found
      #include "config.h"
               ^~~~~~~~~~
      1 error generated.
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pdftopng
Failed to build pdftopng
ERROR: Could not build wheels for pdftopng, which is required to install pyproject.toml-based projects
```

--- a/.github/workflows/run_tests.yml
+++ b/.github/workflows/run_tests.yml
@@ -91,16 +91,6 @@ jobs:
                   restore-keys: |
                       ${{ steps.pre-commit-cache.outputs.result }}-
 
-            - name: Install ghostscript Ubuntu
-              if: matrix.os == 'ubuntu-latest'
-              run: |
-                  sudo apt update
-                  sudo apt install -y ghostscript
-
-            - name: Install ghostscript MacOS
-              if: matrix.os == 'macos-latest'
-              run: brew install ghostscript
-
             - name: Install lightgbm
               if: matrix.os == 'macos-latest'
               run: brew install lightgbm

--- a/docs/usage/usage.md
+++ b/docs/usage/usage.md
@@ -29,7 +29,6 @@ ep.pl.cool_fancy_plot()
 
     io.read_csv
     io.read_h5ad
-    io.read_pdf
     io.write
 ```
 

--- a/ehrapy/io/__init__.py
+++ b/ehrapy/io/__init__.py
@@ -1,2 +1,2 @@
-from ehrapy.io._read import df_to_anndata, read_csv, read_h5ad, read_pdf
+from ehrapy.io._read import df_to_anndata, read_csv, read_h5ad
 from ehrapy.io._write import write

--- a/ehrapy/io/_read.py
+++ b/ehrapy/io/_read.py
@@ -4,7 +4,6 @@ import shutil
 from pathlib import Path
 from typing import Iterator
 
-import camelot
 import numpy as np
 import pandas as pd
 from _collections import OrderedDict
@@ -358,7 +357,6 @@ def _do_read_h5ad(filename: Path | Iterator[str]) -> AnnData:
         decoded_adata = _decode_cached_adata(adata, list(adata.uns["columns_obs_only"]))
         return decoded_adata
     return adata
-
 
 
 def _get_non_existing_files(file: Path, download_dataset_name: str, backup_url: str) -> Path:

--- a/ehrapy/io/_read.py
+++ b/ehrapy/io/_read.py
@@ -108,55 +108,6 @@ def read_h5ad(
     return adata
 
 
-def read_pdf(
-    dataset_path: Path | str,
-    index_column: dict[str, str | int] | str | int | None = None,
-    columns_obs_only: dict[str, list[str]] | list[str] | None = None,
-    columns_x_only: dict[str, list[str]] | list[str] | None = None,
-    return_mudata: bool = False,
-    cache: bool = False,
-    backup_url: str | None = None,
-    download_dataset_name: str | None = None,
-) -> AnnData | dict[str, AnnData] | MuData:
-    """Reads or downloads a desired directory of pdf files or a single pdf file.
-
-    Args:
-        dataset_path: Path to the file or directory to read.
-        index_column: The index column of obs. Usually the patient visit ID or the patient ID.
-        columns_obs_only: These columns will be added to obs only and not X.
-        columns_x_only: These columns will be added to X only and all remaining columns to obs. Note that datetime columns will always be added to .obs though.
-        return_mudata: Whether to create and return a MuData object. This is primarily used for complex datasets which require several AnnData files.
-        cache: Whether to write to cache when reading or not.
-        download_dataset_name: Name of the file or directory in case the dataset is downloaded
-        backup_url: URL to download the data file(s) from if not yet existing.
-
-    Returns:
-        An :class:`~anndata.AnnData` object, a :class:`~mudata.MuData` object or a dict with an identifier (the filename, without extension)
-        for each :class:`~anndata.AnnData` object in the dict
-
-    Example:
-        .. code-block:: python
-
-            import ehrapy as ep
-
-            adatas = ep.io.read_pdf("myfile.pdf")
-    """
-    _check_columns_only_params(columns_obs_only, columns_x_only)
-    file: Path = Path(dataset_path)
-    if not file.exists():
-        file = _get_non_existing_files(file, download_dataset_name, backup_url)
-
-    adata = _read_pdf(
-        filename=file,
-        index_column=index_column,
-        columns_obs_only=columns_obs_only,
-        columns_x_only=columns_x_only,
-        return_mudata=return_mudata,
-        cache=cache,
-    )
-    return adata
-
-
 def _read_csv(
     filename: Path,
     sep: str,
@@ -220,40 +171,6 @@ def _read_h5ad(
     # dataset is a single h5ad file
     else:
         return _do_read_h5ad(filename)
-
-
-def _read_pdf(
-    filename: Path,
-    index_column: dict[str, str | int] | str | int | None,
-    columns_obs_only: dict[str, list[str]] | list[str] | None,
-    columns_x_only: dict[str, list[str]] | list[str] | None,
-    return_mudata: bool = False,
-    cache: bool = False,
-    **kwargs,
-) -> AnnData | dict[str, AnnData] | MuData:
-    """Internal interface of the read_pdf method."""
-    if cache and return_mudata:
-        _cache_not_supported()
-    path_cache = settings.cachedir / filename.stem
-    if cache and (path_cache.is_dir() or path_cache.is_file()):
-        raise CacheExistsException(
-            f"{path_cache} already exists. Use the read_h5ad function instead to read from cache!"
-        )
-    elif filename.is_dir():
-        raise UnsupportedDirectoryParsingFormatException(
-            "Can only parse csv, tsv or h5ad files from a directory, not pdf!"
-        )
-    # dataset is a single pdf file
-    else:
-        adata, columns_obs_only = _do_read_pdf(filename, index_column, columns_obs_only, columns_x_only, cache, **kwargs)  # type: ignore
-        # set cache path, since its a single input file which will be stored in (eventually) multiple cache files
-        path_cache = settings.cachedir / filename.stem  # type: ignore
-        if cache:
-            if not path_cache.parent.is_dir():
-                path_cache.parent.mkdir(parents=True)
-            path_cache.mkdir()
-            return _write_cache_dir(adata, path_cache, columns_obs_only, index_column)  # type: ignore
-        return adata
 
 
 def _read_from_directory(
@@ -442,115 +359,6 @@ def _do_read_h5ad(filename: Path | Iterator[str]) -> AnnData:
         return decoded_adata
     return adata
 
-
-def _do_read_pdf(
-    filename: Path | Iterator[str],
-    index_column: dict[str, str] | None = None,
-    columns_obs_only: dict[str, list[str] | None] = None,
-    columns_x_only: dict[str, list[str] | None] = None,
-    cache: bool = False,
-    **kwargs,
-) -> tuple[dict[str, AnnData], dict[str, list[str] | None] | None]:
-    """Read `.pdf`. Since a single pdf can contain multiple tables, those will be read into a dict.
-
-    Currently, ehrapy only supports parsing single pdfs.
-    Consider the following example: "my_tables.pdf" contains three different tables, which may also differ in size.
-
-        .. code-block:: python
-                       import ehrapy as ep
-                       # read pdf
-                       adata_dict = ep.io.read("my_tables.pdf")
-                       print(adata_dict)
-                       # prints
-                       # {
-                       # "my_tables_1": AnnData object with X obs, Y vars.,
-                       # "my_tables_2": AnnData object with A obs, B vars.,
-                       # "my_tables_3": AnnData object with C obs, D vars.
-                       # }
-
-        The example above illustrates, that the different tables will be stored under the index of the
-        order they appeared in the pdf, prefixed by the pdf filename. This ensures uniqueness, especially in cases
-        when multiple pdfs will be read, with multiple tables per file.
-
-        It's also important to note, that this has to be considered when passing "columns_obs_only":
-            .. code-block:: python
-                           import ehrapy as ep
-
-                           # read pdf
-                           adata_dict = ep.io.read("my_tables.pdf", columns_obs_only={"0":["col1ofTable1", "col2OfTable1"],
-                           "1": ["colOfTable2"]})
-                           # this will put col1 and col2 of Table 0 of my_tables.pdf into obs only for this AnnData object
-                           # and col1 of Table 1 of my_tables.pdf into obs only for this respective AnnData object
-
-    Args:
-        filename: File path to the pdf.
-        index_column: Column name of the index column (obs)
-        columns_obs_only: List (per table) of columns which only be stored in .obs, but not in X. Useful for free text annotations.
-        columns_obs_only: List (per table) of columns which only be stored in X, but not in .obs.
-        cache: Whether the data should be written to cache or not
-
-    Returns:
-        A dict of :class:`~anndata.AnnData` objects and the column obs only for each object
-    """
-    # possible extract modes are lattice (default) or stream; any of them may work better than the other one, depending on the data
-    pdf_extract_mode = kwargs.get("pdf_mode")
-    # camelot does not really parses headers in tables correctly; therefore, if there are any headers in the tables, set them as column names
-    header = kwargs.get("pdf_header")
-    # read a table list from the pdf
-    initial_df_list = camelot.read_pdf(
-        str(filename), flavor=pdf_extract_mode if pdf_extract_mode else "lattice", pages="all"
-    )
-    if not initial_df_list:
-        raise PdfParsingError(
-            f"Failed parsing file {filename}. Could not parse any data."
-            f"Consider converting your data files to .csv/.tsv before parsing or pass the "
-            f"guess parameter to the read function, which may improve parsed results!"
-        )
-
-    ann_data_objects = {}
-
-    # one pdf can contain multiple tables, so each of those tables will be one AnnData object
-    for idx, table in enumerate(initial_df_list):
-        df = table.df
-        is_set = isinstance(header, set)
-        # defaults to True, so if header has not been set, assume first row is column names row
-        # if header is a set and table number idx is in header, also assume first row is also column names row
-        if header is None or (is_set and idx in header):
-            # when the entry in top left corner is empty assume, table has header and index names stored in first row/first column
-            first_empty = df[0][0] == ""
-            headers = df.iloc[0][1 if first_empty else 0 :]
-            if first_empty:
-                index = df.iloc[:, :1][1:].iloc[:, 0]
-            else:
-                index = pd.RangeIndex(start=0, stop=len(df.index) - 1)
-            if first_empty:
-                index.name = ""
-            new_values = df.values[1:, 1:] if first_empty else df.values[1:, :]
-            df = pd.DataFrame(new_values, columns=headers, index=index).apply(
-                pd.to_numeric, args=("ignore",)
-            )  # convert all columns of the DataFrame to numeric, if possible
-
-        this_index_column, this_obs_only, this_x_only = _extract_index_and_columns_obs_only(
-            f"{filename.stem}_{idx}", index_column, columns_obs_only, columns_x_only  # type: ignore
-        )
-
-        # handle x only by moving the asymmetric difference to obs only instead
-        if this_x_only:
-            this_obs_only = list(set(df.columns) - set(this_x_only))
-
-        if columns_obs_only or columns_x_only:
-            initial_df, columns_obs_only[idx] = _prepare_dataframe(df, this_obs_only, columns_x_only, cache)  # type: ignore
-            ann_data_objects[f"{filename.stem}_{idx}"] = df_to_anndata(  # type: ignore
-                initial_df, this_obs_only, this_index_column if this_index_column else None
-            )
-        # in case, no columns_obs_only has been passed
-        else:
-            initial_df, _ = _prepare_dataframe(df, None, None, cache)
-            ann_data_objects[f"{filename.stem}_{idx}"] = df_to_anndata(  # type: ignore
-                initial_df, None, this_index_column if this_index_column else None
-            )
-    # return the initial AnnData object
-    return ann_data_objects, columns_obs_only
 
 
 def _get_non_existing_files(file: Path, download_dataset_name: str, backup_url: str) -> Path:
@@ -881,10 +689,6 @@ class BackupURLNotProvidedError(Exception):
 
 
 class MudataCachingNotSupportedError(Exception):
-    pass
-
-
-class PdfParsingError(Exception):
     pass
 
 

--- a/ehrapy/io/_utility_io.py
+++ b/ehrapy/io/_utility_io.py
@@ -2,7 +2,7 @@ from __future__ import annotations
 
 from pathlib import Path
 
-supported_extensions = {"csv", "tsv", "h5ad", "pdf"}
+supported_extensions = {"csv", "tsv", "h5ad"}
 
 
 def _get_file_extension(file_path: Path) -> str:

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -36,7 +36,6 @@ pynndescent = ">=0.5.4"
 scanpy = ">=1.8.2"
 ipython = ">=7.30.1"
 pyhpo = {extras = ["all"], version = ">=3.0.0"}
-camelot-py = {extras = ["base"], version = ">=0.10.1"}
 deep-translator = ">=1.6.1"
 scikit-learn-intelex = {version = ">=2021.5.3", optional = true}
 medcat = {version = ">=1.2.6", optional = true}

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -47,6 +47,7 @@ session-info = ">=1.0.0"
 lifelines = "^0.27.0"
 ipywidgets = "^8.0.2"
 missingno = "^0.5.1"
+tabulate = "^0.9.0"
 
 [tool.poetry.dev-dependencies]
 pytest = ">=6.2.5"

--- a/tests/io/test_read.py
+++ b/tests/io/test_read.py
@@ -7,7 +7,7 @@ import pytest
 
 from ehrapy.anndata.anndata_ext import ColumnNotFoundError
 from ehrapy.core.tool_available import shell_command_accessible
-from ehrapy.io._read import read_csv, read_h5ad, read_pdf
+from ehrapy.io._read import read_csv, read_h5ad
 
 CURRENT_DIR = Path(__file__).parent
 _TEST_PATH = f"{CURRENT_DIR}/test_data_io"
@@ -144,41 +144,6 @@ class TestRead:
         or (os.name == "nt" and not shell_command_accessible(["gswin64c", " -v"])),
         reason="Requires ghostscript to be installed.",
     )
-    def test_read_pdf(self):
-        adata = read_pdf(dataset_path=f"{_TEST_PATH}/test_pdf.pdf")["test_pdf_0"]
-        assert adata.X.shape == (32, 11)
-        assert adata.var_names.to_list() == [
-            "mpg",
-            "cyl",
-            "disp",
-            "hp",
-            "drat",
-            "wt",
-            "qsec",
-            "vs",
-            "am",
-            "gear",
-            "carb",
-        ]
-        assert id(adata.layers["original"]) != id(adata.X)
-
-    @pytest.mark.skipif(
-        (os.name != "nt" and not shell_command_accessible(["gs", "-h"]))
-        or (os.name == "nt" and not shell_command_accessible(["gswin64c", " -v"])),
-        reason="Requires ghostscript to be installed.",
-    )
-    def test_read_pdf_no_index(self):
-        adata = read_pdf(dataset_path=f"{_TEST_PATH}/test_pdf.pdf")["test_pdf_1"]
-        assert adata.X.shape == (6, 5)
-        assert adata.var_names.to_list() == [
-            "Sepal.Length",
-            "Sepal.Width",
-            "Petal.Length",
-            "Petal.Width",
-            "Species",
-        ]
-        assert id(adata.layers["original"]) != id(adata.X)
-
     def test_set_default_index(self):
         adata = read_csv(dataset_path=f"{_TEST_PATH}/dataset_index.csv")
         assert adata.X.shape == (5, 4)

